internal/bminventory/inventory.go

```diff
@@ -4525,7 +4524,7 @@ func (b *bareMetalInventory) V2RegisterHost(ctx context.Context, params installe
 	//re-calculation occurs only when the host is bound to a multi-node
 	//day 1 cluster and more hosts are present.
 	if triggerAutoAssign {
-		if reset_err := b.resetAutoAssignRoles(ctx, tx, cluster, log, true); reset_err != nil {
+		if reset_err := b.resetAutoAssignRoles(ctx, tx, cluster, log, false); reset_err != nil {
```

Comment:
Now you do not need the last parameter at all. But the refresh is still needed in terms of function. The effective roles are erased here and it affects the cluster state because now many validation may fail again. But the UI does not receive any indications for that and also the monitor may not run them again because the state of the installation has not change

Reply:
The monitor will run again because there is a new host.